From: Rafael Laboissiere <rafael@debian.org> 
Subject: [PATCH] Use FirstName or ForName for Medline


Optionally use FirstName instead of ForeName tag in the Author record, which is
the correct SGML tag name according to the NCBI Medline format
specification (see http://xml.coverpages.org/ncbiPubFormat.html).

 -- Rafael Laboissiere <rafael@debian.org>  Wed, 26 Nov 2008 13:34:04 +0100

Signed-off-by: David Bremner <bremner@unb.ca>


---
 lib/medin.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/medin.c b/lib/medin.c
index f7efe91..1b42f02 100644
--- a/lib/medin.c
+++ b/lib/medin.c
@@ -201,8 +201,8 @@ medin_abstract( xml *node, fields *info )
 /* <AuthorList CompleteYN="Y">
  *    <Author>
  *        <LastName>Barondeau</LastName>
- *        <ForeName>David P</ForeName>
- *        <Initials>DP</Initials>
+ *        <FirstName>David P</FirstName>
+ *        (or <ForeName>David P</ForeName>)
  *    </Author>
  * </AuthorList>
  */
@@ -216,7 +216,7 @@ medin_author( xml *node, newstr *name )
 			newstr_prepend( name, node->value->data );
 		}
 		else newstr_strcat( name, node->value->data );
-	} else if ( xml_tagexact( node, "ForeName" ) ) {
+	} else if ( xml_tagexact( node, "FirstName" ) || xml_tagexact( node, "ForeName" ) ) {
 		p = node->value->data;
 		while ( p && *p ) {
 			if ( name->len ) newstr_addchar( name, '|' );
-- 
tg: (916e574..) feature/medline-tagname (depends on: upstream)
